src/main/java/com/tictactoe/HumanPlayer.java

```diff
@@ -1,27 +1,42 @@
 package com.tictactoe;
 
 public class HumanPlayer {
-    private String symbol;
+    private String input;
     private UserInput receiver;
-    
-    public HumanPlayer(String symbol, UserInput receiver) {
-        this.symbol = symbol;
-        this.receiver = receiver;
+    private UserInputValidator validator = new UserInputValidator();
+    private String symbol;
+
+    public HumanPlayer(UserInput receiver, String symbol) {
+       this.receiver = receiver;
+       this.symbol = symbol;
+    }
+
+    public String getInput() {
+        this.input = this.receiver.obtainInput();
+        return this.input;
     }
 
     public String getSymbol() {
-        return symbol;
+        return this.symbol;
     }
 
-    public void makeMove(Board board) {
-        board.drawGridWithOnlyCellNumberId();
-        int cellSelection = receiver.obtainValidCellSelection();
-        int cellIndex = cellSelection - 1;
-        if (board.getSymbol(cellIndex).equals(" ")) {
-            board.insertSymbol(symbol, cellIndex);
-        } else {
-            System.out.println("Cell already occupied. Please select an empty cell.");
-            makeMove(board);
+    public int obtainValidCellSelection() {
+        int cell;
+        System.out.println("Enter your cell selection (1 - 9): ");
+        while (true)  {
+            try {
+                String selection = this.getInput();
+                cell = Integer.parseInt(selection);
+                break;
+            } catch (NumberFormatException e) {
+                System.out.println("Expected a number. Enter a number between 1 and 9: ");
+            }
+        }
+        boolean isValid = validator.validateCellSelection(cell);
```

Comment:
See my comment from above. Shouldn't the check to see if a cell is taken be included in `isValid`?